DatabindingWithRecyclerViewKotlin/app/src/main/java/com/enpassio/databindingwithrecyclerviewkotlin/ui/ProductAdapter.kt

```diff
@@ -12,28 +12,32 @@ import com.enpassio.databindingwithrecyclerviewkotlin.model.Product
 /**
  * Created by Greta Grigutė on 2018-12-23.
  */
-class ProductAdapter internal constructor(private val mProductList: List<Product>?, private val mListener: ProductItemClickListener) : RecyclerView.Adapter<ProductAdapter.ProductViewHolder>() {
+class ProductAdapter internal constructor(
```

Comment:
@OyaCanli can you please share why are we using `internal constructor` here? Won't a simple primary constructor do the job? I'm not very sure what this would bring so can you please share what additional benefit it'd bring?